### PR TITLE
python: move to traditional mapping

### PR DESCRIPTION
**Pull request by ldelossa**
This moves python matching to the tradtional code flow.

I discovered that without these changes python notifications will not
work correctly.

The Quay team was also made aware that python matching was inconsistent
when utilizing the db filtering method.

This pull results in fewer advisories being munged.

Before:

10:20AM DBG pyupio/updater.go:335 > malformed range component=pyupio/db.Vulnerabilities id=pyup.io-34980 name=bzip spec=>0 updater=pyupio
10:20AM DBG pyupio/updater.go:335 > malformed range component=pyupio/db.Vulnerabilities id=pyup.io-25846 name=html5lib spec=>1.0,<1.0b8 updater=pyupio
10:20AM WRN pyupio/updater.go:290 > malformed database entry component=pyupio/db.Vulnerabilities spec=<2.15.1,>3.0.0,<3.0.0b9 updater=pyupio
10:20AM DBG pyupio/updater.go:335 > malformed range component=pyupio/db.Vulnerabilities id=pyup.io-34982 name=django-server spec=>0 updater=pyupio
10:20AM DBG pyupio/updater.go:335 > malformed range component=pyupio/db.Vulnerabilities id=pyup.io-34986 name=urlib3 spec=>0 updater=pyupio
10:20AM DBG pyupio/updater.go:335 > malformed range component=pyupio/db.Vulnerabilities id=pyup.io-38586 name=keystone spec=>0 updater=pyupio
10:20AM DBG pyupio/updater.go:335 > malformed range component=pyupio/db.Vulnerabilities id=pyup.io-34984 name=setup-tools spec=>0 updater=pyupio
10:20AM DBG pyupio/updater.go:335 > malformed range component=pyupio/db.Vulnerabilities id=pyup.io-37740 name=openstack-keystone spec=>0 updater=pyupio
10:20AM DBG pyupio/updater.go:335 > malformed range component=pyupio/db.Vulnerabilities id=pyup.io-34985 name=telnet spec=>0 updater=pyupio
10:20AM DBG pyupio/updater.go:335 > malformed range component=pyupio/db.Vulnerabilities id=pyup.io-34978 name=acqusition spec=>0 updater=pyupio
10:20AM DBG pyupio/updater.go:335 > malformed range component=pyupio/db.Vulnerabilities id=pyup.io-38694 name=jeIlyfish spec=>0 updater=pyupio
10:20AM DBG pyupio/updater.go:335 > malformed range component=pyupio/db.Vulnerabilities id=pyup.io-34979 name=apidev-coop spec=>0 updater=pyupio
10:20AM DBG pyupio/updater.go:335 > malformed range component=pyupio/db.Vulnerabilities id=pyup.io-33167 name=zodb spec=>=3.9,<3.9.0c2 updater=pyupio
10:20AM DBG pyupio/updater.go:335 > malformed range component=pyupio/db.Vulnerabilities id=pyup.io-36760 name=kubernetes spec=>=9.0,<9.0.0a1 updater=pyupio
10:20AM DBG pyupio/updater.go:335 > malformed range component=pyupio/db.Vulnerabilities id=pyup.io-34981 name=crypt spec=>0 updater=pyupio
10:20AM DBG pyupio/updater.go:335 > malformed range component=pyupio/db.Vulnerabilities id=pyup.io-34987 name=urllib spec=>0 updater=pyupio
10:20AM DBG pyupio/updater.go:335 > malformed range component=pyupio/db.Vulnerabilities id=pyup.io-37745 name=nova spec=>0 updater=pyupio
10:20AM DBG pyupio/updater.go:335 > malformed range component=pyupio/db.Vulnerabilities id=pyup.io-25729 name=django spec=>=1.6,<1.6-beta-4 updater=pyupio
10:20AM DBG pyupio/updater.go:335 > malformed range component=pyupio/db.Vulnerabilities id=pyup.io-25714 name=django spec=>=1.9,<1.9rc2 updater=pyupio
10:20AM DBG pyupio/updater.go:335 > malformed range component=pyupio/db.Vulnerabilities id=pyup.io-25715 name=django spec=>=1.8,<1.8b2 updater=pyupio
10:20AM DBG pyupio/updater.go:335 > malformed range component=pyupio/db.Vulnerabilities id=pyup.io-25713 name=django spec=>=1.8,<1.8c1 updater=pyupio
10:20AM DBG pyupio/updater.go:335 > malformed range component=pyupio/db.Vulnerabilities id=pyup.io-25721 name=django spec=>=1.10,<1.10rc1 updater=pyupio
10:20AM DBG pyupio/updater.go:335 > malformed range component=pyupio/db.Vulnerabilities id=pyup.io-37739 name=cobbler spec=>0 updater=pyupio
10:20AM DBG pyupio/updater.go:335 > malformed range component=pyupio/db.Vulnerabilities id=pyup.io-34983 name=pwd spec=>0 updater=pyupio
10:20AM DBG pyupio/updater.go:345 > munged bounds on some vulnerabilities 😬 component=pyupio/db.Vulnerabilities count=113 updater=pyupio

After:

10:25AM WRN pyupio/updater.go:270 > malformed database entry component=pyupio/Updater.Parse spec=<0.28.3.gfm.12 updater=pyupio
10:25AM WRN pyupio/updater.go:270 > malformed database entry component=pyupio/Updater.Parse spec=<2.-2 updater=pyupio
10:25AM WRN pyupio/updater.go:270 > malformed database entry component=pyupio/Updater.Parse spec=<1.0.1RC1 updater=pyupio
10:25AM WRN pyupio/updater.go:270 > malformed database entry component=pyupio/Updater.Parse spec=<r5238 updater=pyupio
10:25AM DBG pyupio/updater.go:299 > munged bounds on some vulnerabilities 😬 component=pyupio/Updater.Parse count=4 updater=pyupio

Signed-off-by: ldelossa <ldelossa@redhat.com>

### CODE REVIEW

**Comment by ldelossa** [comment]
```
❯ cctool report -dump quay.io/projectquay/quay:latest
2021/02/26 10:16:17 wrote "quay:latest.manifest.json"
2021/02/26 10:16:17 wrote "quay:latest.index.json"
2021/02/26 10:16:17 wrote "quay:latest.report.json"
quay:latest found flask-cors   3.0.8  pyup.io-38745 (CVE-2020-25032)
quay:latest found jsonpickle   1.2    pyup.io-39319 (CVE-2020-22083)
quay:latest found rsa          4.0    pyup.io-38414 (CVE-2020-13757)
quay:latest found rsa          4.0    pyup.io-38369 (CVE-2020-13757)
quay:latest found urllib3      1.25.8 pyup.io-38834 (CVE-2020-26137)
quay:latest found waitress     1.4.2  pyup.io-37667
quay:latest found pyyaml       5.3    pyup.io-38100 (CVE-2020-1747)
quay:latest found cryptography 2.8    pyup.io-39252
quay:latest found cryptography 2.8    pyup.io-38932 (CVE-2020-25659)
quay:latest found pip          9.0.3  pyup.io-38765 (CVE-2019-20916)
````

**Comment by ldelossa** [comment]
for notifications to work correctly, the omnimatcher must be able to check vulnerabilities against package versions in the Indexer:
https://github.com/quay/claircore/blob/master/pkg/omnimatcher/omnimatcher.go

without these changes we do not depend on application code to do the matching correctly, and this is a requirement for notifications to work.

**Comment by ldelossa** [comment]
The python spec parser requires go1.15 minimum so our 1.14 tests are failing.

File: internal/indexer/postgres/affected_manifests_e2e_test.go [hunk 1]
```diff
@@ -269,6 +269,7 @@ func (e *affectedE2E) IndexManifest(t *testing.T) {
 func (e *affectedE2E) AffectedManifests(t *testing.T) {
 	ctx := zlog.Test(e.ctx, t)
 	for _, vuln := range e.vr.Vulnerabilities {
+		zlog.Debug(ctx).Msgf("vuln: %+v", vuln)
```
[the remark this answers is not in the document]
Comment (ldelossa) [reply internal/indexer/postgres/affected_manifests_e2e_test.go:272]: actually, I think I left this in from debugging, should get ripped out all together.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -65,7 +65,7 @@ jobs:
           --health-retries 5
     strategy:
       matrix:
-        go: ['1.14', '1.15']
```
[the remark this answers is not in the document]
Comment (ldelossa) [reply .github/workflows/main.yml:68]: Wouldn't the stable branch still have the old workflow files which pin it to go.1.14?

[the remark this answers is not in the document]
Comment (ldelossa) [reply .github/workflows/main.yml:68]: oh, you're saying I can't backport this change, like this.

[the remark this answers is not in the document]
Comment (ldelossa) [reply .github/workflows/main.yml:68]: confirmed out of band with @thomasmckay building downstream with go1.15 is acceptable.

**Comment by ldelossa** [comment]
@hdonnay let me double check on your concerns with FixedIn not being filled out. 

My initial inclination is that, the only thing that's going to evaluate python vulns are matchers which are looking for them. Presumably these matchers would understand that the verion specs will not be in FixedIn. All other matchers would not even look at these vulns. 

Will double check to confirm.

**Review by hdonnay — APPROVED** [review]
sgtm

**Review by hdonnay — APPROVED** [review]

**Pull request merged**